src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -2071,6 +2104,45 @@ bool WindowImplX11::processEvent(XEvent& windowEvent)
 
             break;
         }
+
+        // Mouse moved raw
+        case GenericEvent:
+        {
+            if (XGetEventData(m_display, &windowEvent.xcookie)
+                    && windowEvent.xcookie.evtype == XI_RawMotion)
+            {
+                const XIRawEvent* rawEv;
+                rawEv = static_cast<const XIRawEvent*>(windowEvent.xcookie.data);
+
+                const int mouseAxis = 2;
+                int relativeValues[2] = { 0, 0 };
+                int limit = rawEv->valuators.mask_len;
+
+                if (limit > mouseAxis)
+                    limit = mouseAxis;
+                limit *= 8;
```

Comment:
I don't understand why `limit` has to be multiplied with 8.

Reply:
Honestly i don't know anymore, SDL did it in a similiar way. But you are right i should not write past the array and there is no point in iterating past `mouseAxis` if i don't use the values anyway.